### PR TITLE
[TRTLLM-9019][feat] Expose video_pruning_rate as llmargs and fix nano-v2-vl

### PR DESCRIPTION
**Pull request by Wanli-Jiang**
# Feature
[None][fix] Fix int64 overflow in Mamba Triton kernels for large sequences
[None][fix] Fix NemotronH Nano input processor for multimodal inputs
[TRTLLM-9019][feat] Expose video_pruning_rate as llmargs

## Summary by CodeRabbit

* **New Features**
  * Added `video_pruning_rate` parameter to control video frame pruning in multimodal models.
  * New `--video_pruning_rate` CLI option for configuring video pruning during model setup.

* **Bug Fixes**
  * Fixed integer overflow in kernel offset calculations for large inputs.
  * Fixed address calculation overflow in layer normalization operations.

* **Tests**
  * Added test coverage for large input edge cases to prevent numerical overflow.

* **Documentation**
  * Updated configuration examples to reflect new video pruning parameter usage.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Review by Superjomn — APPROVED** [review]
LGTM on the llmapi changes.

**Review by nv-guomingz — APPROVED** [review]

**Review by QiJune — APPROVED** [review]
LGTM

**Review by 2ez4bz — COMMENTED** [review]
Mostly nits, except the comment I left on `serve.py`.

File: tensorrt_llm/commands/serve.py [hunk 1]
```diff
@@ -805,6 +814,10 @@ def _serve_llm():
         llm_args = update_llm_args_with_extra_dict(llm_args,
                                                    llm_args_extra_dict)
 
+        # Add video_pruning_rate if specified via CLI (not None)
+        if video_pruning_rate is not None:
+            llm_args["video_pruning_rate"] = video_pruning_rate
+
```
[the remark this answers is not in the document]
Comment (Wanli-Jiang) [reply tensorrt_llm/commands/serve.py]: updated by put `video_pruning_rate` to `get_llmargs` function to unify the code logics

File: tensorrt_llm/llmapi/llm_args.py [hunk 2]
```diff
@@ -3446,6 +3377,15 @@ class TorchLlmArgs(BaseLlmArgs):
...
         status="prototype")
 
+    video_pruning_rate: Optional[float] = Field(
+        default=None,
+        ge=0.0,
+        lt=1.0,
+        description="Pruning rate for video frames in multimodal models. "
+        "Applied by Efficient Video Sampling (EVS) in NemotronH_Nano_VL_V2. "
+        "None (default) disables EVS, values in [0, 1) enable pruning.",
+        status="beta")
```
[the remark this answers is not in the document]
Comment (Wanli-Jiang) [reply tensorrt_llm/llmapi/llm_args.py]: updated

File: tensorrt_llm/_torch/models/modeling_nemotron_nano.py [hunk 3]
```diff
@@ -706,15 +718,15 @@ def get_num_tokens_per_video(
...
         video: List[Image.Image],
-        video_pruning_ratio: Optional[float] = None,
+        video_pruning_rate: Optional[float] = None,
         **kwargs,
     ):
         # Use VIDEO_PRUNING_RATIO if not explicitly provided
-        if video_pruning_ratio is None:
-            video_pruning_ratio = self.video_pruning_ratio
+        if video_pruning_rate is None:
+            video_pruning_rate = self.video_pruning_rate
```
[the remark this answers is not in the document]
Comment (Wanli-Jiang) [reply tensorrt_llm/_torch/models/modeling_nemotron_nano.py]: not yet. 
The method parameter video_pruning_rate on
  get_num_tokens_per_video() exists in the signature but is never called with a non-None value from outside the class — it always falls back to self.video_pruning_rate (set once at init).

  There's no mechanism for per-request override:
  - SamplingParams doesn't include video_pruning_rate
  - The input processor's __call__ method doesn't accept it per-request
  - The multimodal processing pipeline in inputs/multimodal.py calls get_num_tokens_per_video(video=item)
  without passing the parameter

**Comment by Wanli-Jiang** [comment]
/bot skip --comment "CI tests passed in 30594 and 39378"
